sparql-queries/sparql-queries.js

```diff
@@ -76,66 +99,183 @@ export async function finalizePublications(pubTask) {
 
     const now = new Date();
     const finishedStatus = 'http://themis.vlaanderen.be/id/concept/publication-task-status/success';
-    return await query(`
+    return await update(`
         ${PREFIXES}
-        
-        INSERT DATA {
-           GRAPH ${sparqlEscapeUri(pubTask.graph.value)} {
-                ${sparqlEscapeUri(pubTask.publicationTask.value)}       adms:status     ${sparqlEscapeUri(finishedStatus)} ;
-                                                                        dct:modified     ${sparqlEscapeDateTime(now)} .
-                
-           }
+        DELETE {
+            ${sparqlEscapeUri(pubTask.publicationTask)}       adms:status     ?oldStatus;
+                                                              dct:modified     ?oldDate.
+        }
+        INSERT {
+            ${sparqlEscapeUri(pubTask.publicationTask)}       adms:status     ${sparqlEscapeUri(finishedStatus)};
+                                                              dct:modified     ${sparqlEscapeDateTime(now)}.    
+        }
+        WHERE {
+            GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+                ${sparqlEscapeUri(pubTask.publicationTask)}       a               ext:PublicationTask;      
+                                                                  adms:status     ?oldStatus;
+                                                                  dct:modified     ?oldDate .
+            }
         }
         
         `);
 }
 
 export async function saveHtmlContentToPublicationTask(pubTask, html) {
     // The HTML gets stored via nie:htmlContent to the publication task
-    const queryResult = await query(`
+    await update(`
     ${PREFIXES}
-    INSERT DATA {
-       GRAPH ${sparqlEscapeUri(pubTask.graph.value)} {
-            ${sparqlEscapeUri(pubTask.publicationTask.value)}        nie:htmlContent 	${sparqlEscapeString(html)} .
-       }
+    DELETE {
+        GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+            ${sparqlEscapeUri(pubTask.publicationTask)}        nie:htmlContent 	?oldData.
+        }
+    }
+    INSERT {
+        GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+            ${sparqlEscapeUri(pubTask.publicationTask)}        nie:htmlContent 	${sparqlEscapeString(html)}.
+        }
+    } 
+    WHERE {
+        GRAPH ${sparqlEscapeUri(pubTask.graph)} {
+            ${sparqlEscapeUri(pubTask.publicationTask)}        a                ext:PublicationTask;
+                                                               nie:htmlContent 	?oldData.
```

Comment:
Deze query gaat enkel werken indien de publication-task al `nie:htmlContent` heeft, wat doorgaans niet het geval zal zijn. Deze query kan dus beter opgesplitst worden in een `DELETE … WHERE …` met daarna een `INSERT DATA` query.